### DISCUSSION TITLE
Dangling file handle in `load_model`

### DISCUSSION THREAD

**Discussion by phiott** [openai/whisper#1143] [General]
When calling `load_model` with `in_memory=False`, it looks like we have a dangling file handle in https://github.com/openai/whisper/blob/main/whisper/__init__.py#L83. This could be

```python
with open(download_target, "rb") as f:
    model_bytes = f.read()
```

**Comment by jongwook** [comment]
Yes, it could be modified like that to be clearer. Python will be smart enough to release the handle though, as soon as the reference count becomes zero.

**Reply by phiott** [reply]
Ah true, should be good for CPython. I think it might be an issue for PyPy, but that's probably not relevant here.

 Feel free to close.
